docs/migration-to-NRI.md

```diff
@@ -0,0 +1,212 @@
+
+Replace kubelet's --container-runtime-endpoint=/var/run/cri-resmgr/cri-resmgr.sock with --container-runtime-endpoint=/var/run/crio/crio.sock
+
+Replacing the runtime endpoint on a node that was setup using Kubeadm:
+```console 
+# Get the Kubelet args
+systemctl cat kubelet <- Look for: EnvironmentFile=/.../kubeadm-flags.env
+
+vim /.../kubeadm-flags.env
+  KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/crio/crio.sock --pod-infra-container-image=registry.k8s.io/pause:3.9"
+
+vim /etc/sysconfig/kubelet 
+  KUBELET_EXTRA_ARGS= --container-runtime-endpoint=/var/run/crio/crio.sock <- Remember this aswell
+
+systemctl restart crio
+systemctl restart kubelet
+```
+
+Enable NRI:
+```console
+CRIO_CONF=/etc/crio/crio.conf
+cp $CRIO_CONF $CRIO_CONF.orig
+crio --enable-nri config > $CRIO_CONF
+```
+
+## Build the NRI policies
+
+```console
+git clone https://github.com/containers/nri-plugins.git
+cd nri-plugins
+make
+# Build the images, specify your image repo to easily push the image later.
+make images IMAGE_REPO=my-repo IMAGE_VERSION=my-tag
+```
+
+## Create required CRDs
+
+```console
+kubectl apply -f deployment/base/crds/noderesourcetopology_crd.yaml
+```
+
+## Import the image of the NRI plugin you want to run
+
+### Containerd
+
+```console
+ctr -n k8s.io images import build/images/nri-resmgr-topology-aware-image-*.tar
+```
+
+### CRI-O
+
+See the section [below](#steps-for-a-more-real-life-migration-using-self-hosted-image-repository) for instructions on how to push the images to a registry, then pull from there.
+
+## Deploy the plugin
+
+  ```console
+  kubectl apply -f build/images/nri-resmgr-topology-aware-deployment.yaml
+  ```
+
+## Deploy a test pod
+
+```console
+kubectl run mypod --image busybox -- sleep inf
+kubectl exec mypod  -- grep allowed_list: /proc/self/status
+```
+
+## See the resources the pod got assigned with
+
+```console
+kubectl exec $pod -c $container  -- grep allowed_list: /proc/self/status
+# Output should look similar to the output of CRI-RM
+```
+
+# Steps for a more real-life migration using self-hosted image repository
+
+- Same steps as above for enabling NRI with Containerd/CRI-O and building the images.
+- Push the images built to your repository:
+  ```console
+  # Replace my-repo and my-tag with the IMAGE_REPO and IMAGE_VERSION you specified when building the images with make images
+  docker push my-repo:my-tag
+  ```
+
+- Remember to change the image name & pull policy in the plugins .yaml file to match your registyr and image, ex:
+  ```console
+  vim build/images/nri-resmgr-topology-aware-deployment.yaml
+  ```
+
+- Then deploy the plugin simlarly to the earlier step.
+
+# Migrating existing configuration 
+
+- The ConfigMap used by the ported policies/infra has a different name/naming scheme than the original one used in CRI-RM, ex:
+  - configMapName:
+    ```diff
+    - configmap-name: cri-resmgr-config
+    + configmap-name: nri-resmgr-config
```

Comment:
AFAICT, this has been changed to `nri-resource-policy-config` in the [same commit](https://github.com/containers/nri-plugins/commit/bdbb668186a1ab244aca166564bea1c5bfd7c147) as the annotations.